vumi/transports/smpp/smpp_transport.py

```diff
@@ -427,6 +437,7 @@ def process_submit_sm_event(self, message_id, event_type, remote_id,
                     "Could not retrieve failed message: %s" % (message_id,))
             else:
                 yield self.message_stash.delete_cached_message(message_id)
+                yield self.message_stash.expire_multipart_info(message_id)
```

Comment:
I didn't see a test change for this failure path? I could just be sleepy though.